### PR TITLE
Chall02 Solution

### PR DESCRIPTION
**Pull request by tdawson-42**
I use bitmasking with the bitwise AND operator and the value 0xff (255) to discard bits from the integer values so that r, g and b are all between 0-255, and then use bitshifting to move the bits to the correct position. Since the bitmask ensures rgb channel values are between 0-255 I do not check for input outside of that range since the subject doesn't specify required behaviour in such cases. Finally we format the hex value as a string starting with a '#' using sprintf.

### CODE REVIEW

File: chall02/tdawson.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+char *hv_rgb2hex(int r, int g, int b)
+{
+	char *hex_str;
+
+	if (!(hex_str = malloc(8)))
+		return NULL;
+	unsigned int hex = ((r & 0xff) << 16) + ((g & 0xff) << 8) + (b & 0xff);
```
Comment (jgengo) [inline chall02/tdawson.c:10]: wuw :) bitwise!

where did you get it?

File: chall02/tdawson.c [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	sprintf(hex_str, "#%06x", hex);
```
Comment (jgengo) [inline chall02/tdawson.c:11]: if you are using sprintf, you could have avoid bitwise 

```c
sprint(hex_str, "#%02x%02x%02x", r, g, b);
```

**Review by jgengo — APPROVED** [review]
really clear, and nice you used bitwise!

File: chall02/tdawson.c (hunk 1 above)
Comment (Astridchan) [reply chall02/tdawson.c]: I found it when doing research for FDF but I can't remember where, probably from Stack Overflow (where else 😁)

Comment (tdawson-42) [reply chall02/tdawson.c:10]: I found it when doing research for my FDF but I can't remember where, probably Stack Overflow (where else 😁)

Comment (jgengo) [reply chall02/tdawson.c:10]: Stack Overflow :) the developer toolbox!

**Pull request merged**